src/Microsoft.OData.Client/DeepInsertSaveResult.cs

```diff
@@ -268,6 +329,21 @@ protected override ODataRequestMessageWrapper CreateRequestMessage(string method
             return this.CreateTopLevelRequest(method, requestUri, headers, httpStack, descriptor);
         }
 
+        /// <summary>Read and store response data for the current change</summary>
+        /// <param name="pereq">The completed per request object</param>
+        /// <remarks>This is called only from the async code paths, when the response to the batch has been read fully.</remarks>
+        protected override void FinishCurrentChange(PerRequest pereq)
+        {
+            base.FinishCurrentChange(pereq);
+
+            // This resets the position in the buffered response stream to the beginning
+            // so that we can start reading the response.
+            // In this case the ResponseStream is always a MemoryStream since we cache the async response.
+            this.ResponseStream.Position = 0;

```

Comment:
```suggestion
            this.responseStream.Position = 0;
```
Better to reference the private field here instead of public property